### PR TITLE
[macOS] Add pipx and install yamllint through it

### PR DESCRIPTION
**Pull request by nikita-bykov**
# Description
Added pipx tool on macOS and installed yamllint through it

#### Related issue:
#1657 
## Check list
- [x] Related issue / work item is attached
- [x] Tests are written (if applicable)
- [x] Documentation is updated (if applicable)
- [x] Changes are tested and related VM images are successfully generated

## Builds

- [macOS 10.13](https://github.visualstudio.com/virtual-environments/_build/results?buildId=85193&view=logs&j=80d86624-2c9a-5cad-9434-144b838e69fb)
- [macOS 10.14](https://github.visualstudio.com/virtual-environments/_build/results?buildId=85192&view=logs&j=80d86624-2c9a-5cad-9434-144b838e69fb)
- [macOS 10.15](https://github.visualstudio.com/virtual-environments/_build/results?buildId=85194&view=logs&j=80d86624-2c9a-5cad-9434-144b838e69fb&t=59429f13-1073-5598-b503-957696391418)
- [macOS 11.0](https://github.visualstudio.com/virtual-environments/_build/results?buildId=85190&view=logs&j=80d86624-2c9a-5cad-9434-144b838e69fb)

### CODE REVIEW

File: images/macos/provision/core/pipx-packages.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+export PATH="$PATH:/opt/pipx_bin"
+
+toolset=get_toolset_path
+pipx_packages=$(jq -r ".pipx[] .package" $toolset)
+
+for package in $pipx_packages; do
+    python_version=$(jq -r ".pipx[] | select(.package == \"$package\") .python" $toolset)
+    if [ "$python_version" != "null" ]; then
+        python_path="/opt/hostedtoolcache/Python/$python_version*/x64/bin/python$python_version"
```
Comment (maxim-lobanov) [inline images/macos/provision/core/pipx-packages.sh:11]: We don't have `/opt/hostedtoolcache` on MacOS

File: images/macos/provision/core/pipx-packages.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        echo "Install $package into default python"
+        pipx install $package
+    fi
+
+    # Run tests to determine that the software installed as expected
+    cmd=$(jq -r ".pipx[] | select(.package == \"$package\") .cmd" $toolset)
+    if ! command -v $cmd; then
+        echo "$package was not installed"
+        exit 1
+    fi
```
Comment (maxim-lobanov) [inline images/macos/provision/core/pipx-packages.sh:24]: We don't need these tests here. On MacOS Tests are located separately.

File: images/macos/provision/core/python.sh [hunk 3]
```diff
@@ -7,7 +7,19 @@ echo "Brew Installing Python 3"
...
 echo "Brew Installing Python 2"
-# Create local tap with formula due to python2 formula depreciation 
+# Create local tap with formula due to python2 formula depreciation
 /usr/local/bin/brew tap-new local/python2
 FORMULA_PATH=$(/usr/local/bin/brew extract python@2 local/python2 | grep "Homebrew/Library/Taps")
 /usr/local/bin/brew install $FORMULA_PATH
+
+echo "Installing pipx"
+export PIPX_BIN_DIR=/opt/pipx_bin
+export PIPX_HOME=/opt/pipx
```
Comment (maxim-lobanov) [inline images/macos/provision/core/python.sh:17]: May be `/usr/local`? @al-cheb for any thoughts

File: images/macos/provision/core/pipx-packages.sh (hunk 1 above)
Comment (nikita-bykov) [reply images/macos/provision/core/pipx-packages.sh:11]: Fixed it

File: images/macos/provision/core/pipx-packages.sh (hunk 2 above)
Comment (nikita-bykov) [reply images/macos/provision/core/pipx-packages.sh:24]: Removed it

File: images/macos/provision/core/python.sh (hunk 3 above)
Comment (nikita-bykov) [reply images/macos/provision/core/python.sh:17]: I think `/usr/local` is the better palse, so I moved it to there

File: images/macos/provision/core/commonutils.sh [hunk 4]
```diff
@@ -22,7 +22,6 @@ binst_common_utils=(
     github/gh/gh
     p7zip
     ant
-    yamllint
```
Comment (vsafonkin) [inline images/macos/provision/core/commonutils.sh:25]: We should install yamllint on High Sierra via brew

File: images/macos/software-report/SoftwareReport.Generator.ps1 [hunk 5]
```diff
@@ -97,6 +98,7 @@ if ($os.IsLessThanBigSur) {
 
 $markdown += New-MDList -Style Unordered -Lines @(
     "Pip ${pip3Version}",
+    $pipxVersion,
```
Comment (vsafonkin) [inline images/macos/software-report/SoftwareReport.Generator.ps1:101]: Software report for High Sierra will be contain `Pipx` line, you should add condition for High Sierra.

File: images/macos/provision/core/pipx-packages.sh [hunk 6]
```diff
@@ -0,0 +1,18 @@
+source ~/utils/utils.sh
+
+export PATH="$PATH:/opt/pipx_bin"
+
+toolset=get_toolset_path
+pipx_packages=$(jq -r ".pipx[] .package" $toolset)
```
Comment (dmitry-shibanov) [inline images/macos/provision/core/pipx-packages.sh:6]: You should use 
```
toolset=$(get_toolset_path)
pipx_packages=$(jq -r ".pipx[] .package" "$toolset") 
```
or 
`pipx_packages=$(get_toolset_value ".pipx[] .package")`

**Review by vsafonkin — APPROVED** [review]

File: images/macos/toolsets/toolset-11.0.json [hunk 7]
```diff
@@ -88,5 +88,11 @@
                 "1.15.*"
             ]
         }
+    ],
+    "pipx": [
+        {
+            "package": "yamllint",
+            "cmd": "yamllint"
```
Comment (maxim-lobanov) [inline images/macos/toolsets/toolset-11.0.json:95]: Why do we add cmd field? Looks like it is not used anywhere

File: images/macos/templates/macOS-10.15.json [hunk 8]
```diff
@@ -196,6 +196,13 @@
             "execute_command": "chmod +x {{ .Path }}; {{ .Vars }} pwsh -f {{ .Path }}",
             "scripts": "./provision/core/toolset.ps1"
         },
+        {
+            "type": "shell",
+            "execute_command": "chmod +x {{ .Path }}; {{ .Vars }} {{ .Path }}",
+            "scripts": [
+                "./provision/core/pipx-packages.sh"
```
Comment (maxim-lobanov) [inline images/macos/templates/macOS-10.15.json:203]: Please move this script to the section where `pypy.sh` is located. It doesn't make sense to create the new one

File: images/macos/toolsets/toolset-11.0.json (hunk 7 above)
Comment (maxim-lobanov) [reply images/macos/toolsets/toolset-11.0.json:95]: Also could we iterate by this JSON in tests?

Comment (nikita-bykov) [reply images/macos/toolsets/toolset-11.0.json]: Rework tests

File: images/macos/templates/macOS-10.15.json (hunk 8 above)
Comment (nikita-bykov) [reply images/macos/templates/macOS-10.15.json]: Moved it

**Review by MaksimZhukov — APPROVED** [review]

**Review by maxim-lobanov — APPROVED** [review]

**Review by al-cheb — APPROVED** [review]

**Pull request merged**